v2/internal/controllers/crd_networking_applicationgateway_test.go

```diff
@@ -0,0 +1,389 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"log"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	network2020 "github.com/Azure/azure-service-operator/v2/api/network/v1api20201101"
+	network "github.com/Azure/azure-service-operator/v2/api/network/v1api20220701"
+	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1api20200601"
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+func Test_Networking_ApplicationGateway_HTTPS_Listener_CRUD(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+	appGatewayName := "appgatewaytest"
+	vnet := newVNet(tc, testcommon.AsOwner(rg), []string{"10.0.0.0/8"})
+	subnet := &network2020.VirtualNetworksSubnet{
+		ObjectMeta: tc.MakeObjectMeta("subnet"),
+		Spec: network2020.VirtualNetworks_Subnet_Spec{
+			Owner:         testcommon.AsOwner(vnet),
+			AddressPrefix: to.Ptr("10.0.0.0/24"),
+			AzureName:     "AzureAppGtwSubnet", // The subnet must have this name in Azure for Bastion to work
+		},
+	}
+	publicIPAddress := newPublicIp(tc, testcommon.AsOwner(rg))
+	tc.CreateResourceAndWait(publicIPAddress)
```

Comment:
minor, you can combined all of these into a single create+wait

`tc.CreateResourcesAndWait(publicIPAddress, vnet, subnet)`